src/utils.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+module.exports = {
+  getProtocol: function(url) {
+    var link = document.createElement('a');
+    link.href = url;
+    return link.protocol;
```

Comment:
Yeah, could do - though was thinking ahead we're probably going to want to extend this to get hosts later.